app/models/SearchAfterCache.scala

```diff
@@ -0,0 +1,52 @@
+package models
+
+import java.time.{Duration, LocalDateTime}
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import javax.inject.Inject
+import play.api.inject.SimpleModule
+import play.api.{Logging, inject}
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+class CacheModule extends SimpleModule(inject.bind[SearchAfterCache].toSelf.eagerly())
+
+object SearchAfterCache extends Logging {
```

Comment:
@JarrodBaker be careful with this. we should not implementing this ourselves. Woud you mind check this already way to do it as Play already provide this cache mechanism for you (Caffeine) https://www.playframework.com/documentation/2.8.x/ScalaCache

Reply:
Using internal simple cache for caffeine would be enough

Reply:
Ehcache will be obsoleted and caffeine is the new one

Reply:
Thanks for the tip: I've deleted by home grown stuff and have implemented using Caffeine.